src/averageAges.js

```diff
@@ -34,7 +40,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let filteredWomen = [];
+
+  if (withChildren) {
+    filteredWomen = people.filter(
+      person => person.sex === 'f' && people.some(
+        human => human.mother === person.name));
+  } else {
+    filteredWomen = people.filter(person => person.sex === 'f');
+  }
```

Comment:
Так само краще в одному фільтрі
```suggestion
  filteredWomen = people.filter(
    withChildren
    ? ...yourCode
    : yourCode
  )
```